crates/bevy_ecs/src/entity/clone_entities.rs

```diff
@@ -175,29 +185,51 @@ impl<'w> EntityCloneBuilder<'w> {
             filter_allows_components,
             filter: Arc::new(filter),
             clone_handlers_overrides: Arc::new(clone_handlers_overrides),
+            move_components,
         }
         .clone_entity(world);
 
         world.flush_commands();
     }
 
+    /// Allows for a scoped mode where any changes to the filter that allow/deny components
+    /// will also allow/deny those components' required components, recursively.
+    pub fn with_required_components(
+        &mut self,
+        builder: impl FnOnce(&mut EntityCloneBuilder) + Send + Sync + 'static,
+    ) -> &mut Self {
+        self.attach_required_components = true;
+        builder(self);
+        self.attach_required_components = false;
+        self
+    }
+
+    /// Sets the cloner to remove any components that were cloned,
+    /// effectively moving them from the source entity to the target.
+    ///
+    /// This only applies to components that are allowed through the filter
+    /// at the time [`EntityCloneBuilder::clone_entity`] is called.
+    pub fn enable_move_on_clone(&mut self) -> &mut Self {
+        self.move_components = true;
+        self
+    }
+
+    /// The default setting for the cloner. The source entity will keep
```

Comment:
The bit about this being the default should be in the second line, not the first.